### PR TITLE
Stop providing author inherent data

### PR DESCRIPTION
**Pull request by JoshOrndorff**
DO NOT MERGE until after nimbus 0.9 has been tagged.

This PR is a followup to #15. It simplifies the client-side code by no longer providing any real inherent data. Previously it provided the author id, but since #15, the data supplied there has not been used, and the pre-runtime digest is used instead.

### CODE REVIEW

File: parachain-template/node/src/service.rs [hunk 1]
```diff
@@ -400,7 +395,7 @@ pub async fn start_parachain_node(
 							)
 						})?;
 
-						let author = nimbus_primitives::InherentDataProvider::<NimbusId>(author_id);
+						let author = nimbus_primitives::InherentDataProvider;
 
 						Ok((time, parachain_inherent, author))
```
Comment (librelois) [inline parachain-template/node/src/service.rs]:
```suggestion
						Ok((time, parachain_inherent, nimbus_inherent))
```

File: parachain-template/node/src/service.rs [hunk 2]
```diff
@@ -400,7 +395,7 @@ pub async fn start_parachain_node(
 							)
 						})?;
 
-						let author = nimbus_primitives::InherentDataProvider::<NimbusId>(author_id);
+						let author = nimbus_primitives::InherentDataProvider;
```
Comment (librelois) [inline parachain-template/node/src/service.rs]:
```suggestion
let nimbus_inherent = nimbus_primitives::InherentDataProvider;
```

**Review by librelois — CHANGES_REQUESTED** [review]

**Review by librelois — APPROVED** [review]

**Comment by JoshOrndorff** [comment]
@librelois @nbaztec I believe this one is ready. Is there anything blocking it that you know of?

**Comment by librelois** [comment]
@JoshOrndorff yes we can merge this PR now, we just forgot because we were constantly busy with other things!

**Pull request merged**